src/common/pa_converters.c

```diff
@@ -935,13 +935,31 @@ static void Int32_To_Int24_Dither(
     void *sourceBuffer, signed int sourceStride,
     unsigned int count, struct PaUtilTriangularDitherGenerator *ditherGenerator )
 {
-    (void) destinationBuffer; /* unused parameters */
-    (void) destinationStride; /* unused parameters */
-    (void) sourceBuffer; /* unused parameters */
-    (void) sourceStride; /* unused parameters */
-    (void) count; /* unused parameters */
     (void) ditherGenerator; /* unused parameters */
-    /* IMPLEMENT ME */
+
+    PaInt32 *src = (PaInt32*)sourceBuffer;
+    signed char *dest = (signed char*)destinationBuffer;
+    signed char *scaledDitherResult = (signed char*)destinationBuffer;
+    PaInt32 dither;
+
+    while ( count-- )
+    {
+        /* REVIEW */
+        dither = PaUtil_Generate16BitTriangularDither(ditherGenerator);
+        *scaledDitherResult = (signed char) ((((*src) >> 1) + dither) >> 7);
+
+#if defined(PA_LITTLE_ENDIAN)
+        dest[0] = (unsigned char)(*scaledDitherResult >> 8);
```

Comment:
You already shifted by eight up above so you can shift by eight less when doing char writes.